deploy/cmd/create.go

```diff
@@ -0,0 +1,341 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/avast/retry-go"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ctypes "github.com/ovrclk/akash/provider/cluster/types"
+	gateway "github.com/ovrclk/akash/provider/gateway/rest"
+	dcli "github.com/ovrclk/akash/x/deployment/client/cli"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	// FlagTimeout represents max amount of time for lease status checking process
+	FlagTimeout = "timeout"
+	// FlagTick represents time interval at which lease status is checked
+	FlagTick = "tick"
+)
+
+func retryIfGatewayClientResponseError(err error) bool {
+	_, isClientErr := err.(gateway.ClientResponseError)
+	return isClientErr
+}
+
+var errDeployTimeout = errors.New("Timed out while trying to deploy")
+
+// createCmd represents the create command
+func createCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [sdl-file]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Create a deployment on the akash network",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			timeoutDuration := viper.GetDuration(FlagTimeout)
+			endAt := time.Now().Add(timeoutDuration)
+			ctx, cancel := context.WithDeadline(context.Background(), endAt)
```

Comment:
i think it is better to use `cmd.Context()`

Reply:
Good point, I'll give that a try